tests/unit/test_charm.py

```diff
@@ -20,7 +19,7 @@ def setUp(self):
         self.harness = testing.Harness(MagmaAccessGatewayOperatorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
-        self.charm = typing.cast(MagmaAccessGatewayOperatorCharm, self.harness.charm)
+        self.charm = self.harness.charm
```

Comment:
@ghislainbourgeois I've removed this because it was flagged as "redundant", let me know if there's a reason for keeping it and I'll revert the change :)

Reply:
I added this to fix a complaint of `mypy`, so if it is not complaining now, the underlying issue has been fixed.